src/render/tutorials/chrome-plugin/2_2.html.md

```diff
@@ -0,0 +1,495 @@
+
+If we were to leave the configuration like this we would create a REST API. We want to render a page though, so we need another node in our task. When the first node succeeds (so user data is fetched), we want to call a node that renders a page. This is the updated code for the `user-task`:
+
+```hocon
+user-task {
+  action = fetch-user-info
+  on._success {
+    action = te-hbs
+  }
+}
+```
+
+The `on` object contains all transitions (graph edges) we want to be able to handle. The format looks like this:
+
+```hocon
+on {
+  transition1 = node1
+  transition2 = node2
+  // ...
+}
+```
+
+In this case we define that after a node with action `fetch-user-info` Knot.x should call a node with action `te-hbs`. This action will render the page using handlebars. This is how it looks like:
+
+```hocon
+te-hbs {
+  factory = knot
+  config.address = knotx.knot.te.handlebars
+}
+```
+
+Perhabs not what you were expecting. The `knot` factory allows for registering actions on the [Vert.x Event Bus](https://vertx.io/docs/vertx-core/java/#event_bus), thus providing more scalability. Here's how the actual handlebars configuration looks like:
+
+```hocon
+address = knotx.knot.te.handlebars
+
+engine {
+  factory = handlebars
+  config = {
+    cacheSize = 1000
+  }
+}
+```
+
+The entire configuration for the first snippet:
+
+```hocon
+// tasks.conf
+
+user-task {
+  action = fetch-user-info
+  on._success {
+    action = te-hbs
+  }
+}
+
+// actions.conf
+
+fetch-user-info {
+  factory = http
+  config.endpointOptions {
+    path = /user
+    host = webapi
+    port = 8080
+  }
+}
+te-hbs {
+  factory = knot
+  config.address = knotx.knot.te.handlebars
+}
+```
+
+Now the User Details section of the first snippet should render like this:
+
+![Rendered User Details](/img/blog/chrome-extension/filled-user-details.png)
+
+Here's when the chrome extension comes in. If we have [Fragment HTML Body Writer](https://github.com/Knotx/knotx-fragments/tree/master/handler/consumer/html) configured 
+adequately, the extension can read data about fragments from the HTML page. Assuming the extension is installed, let's open the inspector and go to the "Knot.x" tab. We can see a list of fragments on the page:
+
+
+![Rendered User Details](/img/blog/chrome-extension/list-of-fragments.png)
+
+There are two fragments because we have two snippets in HTML. We haven't defined a task for the second one thus it displayes a grey square next to it. It means it wasn't processed. Ignore it for now and focus on the first one. It's green so it ended with a success. There's also a randomly generated ID, a type ("snippet" is the only possible fragment type in an HTML page), and the time it took to process the fragment. If you click on the arrow pointing down next to the ID, you should see a list of top level HTML elements in the snippet. In this case it's just one div. Hover over it to highlight the div on the page.

```

Comment:
> "snippet" is the only possible fragment type in an HTML page

this is not true, we can define more types such as: `hbs-snippet` or `pebble-snippet`

Then in Fragment HTML Body Writer we configure the [fragmentType](https://github.com/Knotx/knotx-fragments/tree/master/handler/consumer/html#how-to-configure) option.